packages/taco/src/taco-client.ts

```diff
@@ -0,0 +1,322 @@
+      })();
+    }
+    return TacoClient.initializationPromise;
+  }
+
+  /**
+   * Create a new TacoClient instance
+   *
+   * @param {TacoClientConfig} config - Configuration for the TacoClient
+   * @throws {Error} If configuration is invalid
+   */
+  constructor(config: TacoClientConfig) {
+    // Validate configuration using TacoConfig
+    const result = TacoConfigValidator.validateFast(config);
+    if (!result.isValid) {
+      throw new Error(`Invalid configuration: ${result.errors.join(', ')}`);
+    }
+
+    this.config = config;
+
+    TacoClient.initialize();
+  }
+
+  /**
+   * Fully validate the configuration including network provider checks
+   *
+   * This method performs comprehensive validation including:
+   * - Domain and ritual ID validation
+   * - Provider/signer configuration validation
+   * - Network compatibility check (calls provider to verify chain ID matches domain)
+   *
+   * @returns {Promise<ValidationResult>} Promise resolving to validation result with isValid boolean and errors array
+   * @throws {Error} If configuration validation fails
+   *
+   * @example
+   * ```typescript
+   * try {
+   *   const result = await tacoClient.validateConfig();
+   *   console.log('Configuration is valid:', result.isValid);
+   * } catch (error) {
+   *   console.error('Configuration validation failed:', error.message);
+   * }
+   * ```
+   */
+  async validateConfig(): Promise<ValidationResult> {
+    const validationResult = await TacoConfigValidator.validateFull(
+      this.config,
+    );
+    if (!validationResult.isValid) {
+      throw new Error(
+        `Invalid configuration: ${validationResult.errors.join(', ')}`,
+      );
+    }
+    return validationResult;
+  }
+
+  /**
+   * Encrypt data with the given access condition
+   *
+   * @param {string | Uint8Array} data - String or Uint8Array to encrypt
+   * @param {Condition} accessCondition - Access condition for decryption
+   * @returns {Promise<ThresholdMessageKit>} Encrypted message kit
+   * @throws {Error} If encryption fails
+   *
+   * @example
+   * ```typescript
+   * const messageKit = await tacoClient.encrypt('Hello, secret!', condition);
+   * ```
+   */
+  async encrypt(
+    data: string | Uint8Array,
+    accessCondition: Condition,
+  ): Promise<ThresholdMessageKit> {
+    await TacoClient.initialize();
+
+    try {
+      const messageKit = await encrypt(
+        (this.config as TacoClientEthersConfig).ethersProvider ||
+          (this.config as TacoClientViemConfig).viemClient,
+        this.config.domain,
+        data,
+        accessCondition,
+        this.config.ritualId,
+        (this.config as TacoClientEthersConfig).ethersSigner ||
+          (this.config as TacoClientViemConfig).viemSignerAccount,
+      );
+
+      return messageKit;
+    } catch (error) {
+      const errorMessage =
+        error instanceof Error ? error.message : String(error);
+      throw new Error(`TACo encryption failed: ${errorMessage}`);
```

Comment:
We shouldn't modify the thrown error, because then the encrypt API and this API would return different errors. I think we should just remove the try-catch.

Same comment for the other functions.